### ISSUE TITLE
fix numbering in PDBs when negatives are actually used

### ISSUE THREAD

**Issue by swails** [ParmEd/ParmEd#827]
From @dacase:

Hi Jason:

I can't remember if we addressed this before, but there is a problem
in the PDBFile.write() method in pdb.py when the input residue numbers
are negative or zero.  (I don't see a fix even in the gitub.com/amber-md
version.)

Existing code around line 821:
```python
                    # Figure out the serial numbers we want to print
                    if renumber:
                        anum = (atom.idx + 1 + nmore)
                        rnum = (res.idx + 1)
                    else:
                        anum = (pa.number or last_number + 1)
                        rnum = (atom.residue.number or last_rnumber + 1)
                    anum = anum - anum // 100000 * 100000
                    rnum = rnum - rnum // 10000 * 10000
```
This fails if renumber=False and atom.residue.number is < 1.  My suggestion:
```python
                    if renumber:
                        anum = (atom.idx + 1 + nmore) % 100000
                        rnum = (res.idx + 1) % 10000
                    else:
                        anum = pa.number
                        rnum = atom.residue.number
                        # optional--depends on where you think the atom
                        #   and residue numbers originally came from:
                        # This should prevent both positive and negative
                        #   numbers from overflowing.  But would not be
                        #   needed the the original values came from a
                        #   load_file() operation.
                        if anum > 0:
                            anum = anum % 100000
                        else
                            anum = anum % 10000
                        if rnum > 0:
                            rnum = rnum % 10000
                        else
                            rnum = rnum % 1000
```
Basically, if renumber=False, you want to print out whatever was there
originally, not try to guess with things like last_rnumber+1 (which gives
every atom a different residue number if atom.residue.number=0).

I've actually run across three PDB files recently that have negative or
zero residue numbers; so this is rare but not unknown.

Apologies if I'm missing something here, or if there is some other
interpretation of renumber=False that is intended.

...dac

**Comment by swails** [comment]
I'll have to think about this -- the value -1 is used as a "no atom number was assigned here" (e.g., if you add atoms to a system or combine them or something), so my intent was to just increment (there's no "right" answer here).

This seems less common than actually *having* negative indexes though, so I should probably adopt the above suggestion.

**Comment by swails** [comment]
Fixed in #838
